packages/ember-data/lib/serializers/rest_serializer.js

```diff
@@ -487,20 +490,22 @@ var RESTSerializer = JSONSerializer.extend({
     var payload = this.normalizePayload(rawPayload);
 
     for (var prop in payload) {
-      var typeName = this.typeForRoot(prop);
-      if (!store.modelFactoryFor(typeName, prop)) {
-        Ember.warn(this.warnMessageNoModelForKey(prop, typeName), false);
-        continue;
-      }
-      var type = store.modelFor(typeName);
-      var typeSerializer = store.serializerFor(type);
+      if (prop !== "meta") {
```

Comment:
As I'm looking at it, `extractMeta` doesn't just remove the meta key, it actually goes and sets the metadata on the store. Which is fine, it means the store could be simpler, but the downside is that it requires a type, and we don't know the type at this point. So, I'm going to replace the code in store with a call to `serializer.extractMeta`, before we call `serializer.pushPayload`, since that's the only place where we have all the ingredients required for a call to `JSONSerializer.extractMeta`